File: src/main/java/no/novari/configuration/model/configuration/entities/collection/InstanceCollectionReference.java
package no.novari.configuration.model.configuration.entities.collection;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.validation.constraints.NotBlank;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.extern.jackson.Jacksonized;

@Getter
@Builder
@Jacksonized
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class InstanceCollectionReference {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonIgnore
    @Setter(AccessLevel.NONE)
    private long id;

    private int index;

    @NotBlank
    private String reference;

}

File: src/main/kotlin/no/novari/flyt/configuration/model/configuration/entities/collection/InstanceCollectionReference.kt
package no.novari.flyt.configuration.model.configuration.entities.collection

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.validation.constraints.NotBlank

@Entity
class InstanceCollectionReference(
    @field:Id
    @field:GeneratedValue(strategy = GenerationType.IDENTITY)
    @field:JsonIgnore
    var id: Long = 0,
    var index: Int = 0,
    @field:NotBlank
    @field:Column(columnDefinition = "text")
    var reference: String? = null,
) {
    companion object {
        @JvmStatic
        fun builder(): Builder = Builder()
    }

    class Builder {
        private var id: Long = 0
        private var index: Int = 0
        private var reference: String? = null

        fun id(id: Long) = apply { this.id = id }

        fun index(index: Int) = apply { this.index = index }

        fun reference(reference: String?) = apply { this.reference = reference }

        fun build(): InstanceCollectionReference =
            InstanceCollectionReference(id = id, index = index, reference = reference)
    }
}
